### PR TITLE
Pagetree drag n drop copy page

### PR DESCRIPTION
**Pull request by vxsx**
video: http://d.pr/v/CESu

- [x] basic functionality (cmd/ctrl+drag)
- [ ] advanced (nested trees etc) (apparently needs backend help)
- [x] integration tests for simple cases
- [x] integration tests for complex cases (pasting a page into its descendants)

additionally fixes a layout bug in safari

### CODE REVIEW

**Comment by vxsx** [comment]
@mkoistinen this kind of functionality is currently acting weird http://d.pr/v/1eLsw so I need your help :)
there is a failing integration test added in
`gulp tests:integration --tests=pagetree-drag-n-drop-copy`

File: cms/tests/frontend/integration/pagetree-drag-n-drop-copy.js [hunk 1]
```diff
@@ -0,0 +1,485 @@
...
+
+                // usually to drag stuff in the iframe you have to calculate the position of the frame
+                // and then the position of the thing inside frame, but here sideframe is opened at 0, 0
+                // so this should be enough
+                drop = this.getElementBounds(
+                    xPath('//a[contains(@class, "jstree-anchor")][contains(text(), "Homepage")]')
+                );
+
+                this.mouse.down(xPath('//a[contains(@class, "jstree-anchor")][contains(text(), "Homepage")]'));
+                this.mouse.move(drop.left + drop.width / 2, drop.top + drop.height - 3);
```
Comment (FinalAngel) [inline cms/tests/frontend/integration/pagetree-drag-n-drop-copy.js]: this looks a bit like voodoo to me, not sure whats happening there.

**Comment by FinalAngel** [comment]
FE LGTM

File: cms/tests/frontend/integration/pagetree-drag-n-drop-copy.js (hunk 1 above)
Comment (vxsx) [inline cms/tests/frontend/integration/pagetree-drag-n-drop-copy.js]:
```
+------------------------+
| Homepage               +
+-----------X------------+
```
there X is a cursor position when dropping (+ 3px on Y)

**Comment by czpython** [comment]
@vxsx 
All existing BE tests are passing locally and the provided use case also works now.
Pending is to write some BE regression tests.

**Comment by czpython** [comment]
Just FYI, I noticed a lot of lag with in the FE when dragging pages.

**Pull request closed without merging**